mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -1390,14 +1418,26 @@ def GPU_GPUModuleOp : GPU_Op<"module", [
       gpu.module_end
     }
     ```
+
+    Modules can have optional spatial extents. Currently you can specify the
+    required subgroup size, required workgroup size, and the max workgroup size.
+    It is invalid to specify both the required workgroup size and the max
+    workgroup size.
+
+    ```
+    gpu.module @symbol_name extents<#gpu.spatial_extents<reqdSubgroupSize=32, maxWorkgroupSize=128>> {}
```

Comment:
```suggestion
    gpu.module @symbol_name extents<#gpu.spatial_extents<reqdSubgroupSize=32, maxWorkgroupSize=[128]>> {}
```

That is not a list, right? Does it work like that?

Reply:
seems to work fine for me

Reply:
Amazing

Reply:
After converting from `std::optional<DenseI32ArrayAttr>` to `DenseI32ArrayAttr`, this no longer works!